### ISSUE TITLE
Viewer Missing Print Button

### ISSUE THREAD

**Issue by Muhammad-Radwan** [FastReports/FastReport#406]
I'm using the original viewer application to preview and print prepared reports. In my application I'm using a process to start the application which I already associated the '.fpx' file with. It works fine except for that I have to press 'CTRL+P' in order to print the current view because there is no 'Print' button available. And if there is a better way to preview and print the reports programmatically, please guide me to it.

**Comment by Detrav** [comment]
Are you using Community version? Community does not have a print function

**Comment by Muhammad-Radwan** [comment]
@Detrav Yes, I'm using the community version. Do you mean that I have to switch to the paid version in order to print?

**Comment by Detrav** [comment]
Community only for design a report, cause open source version does not have its own designer, the team added a special version of free designer, but wht a many restrictions.

If you need to print from a desktop, then this is a paid version or print through a browser.

**Comment by Detrav** [comment]
The limitation of community is technical, most likely, this is a bug that the printed code is available in the community assembly 😅

**Comment by Muhammad-Radwan** [comment]
@Detrav So I shall shut my mouth and tell no body about it lol....

**Comment by Muhammad-Radwan** [comment]
I discovered a bug, they should reward me.. you know how? enable printing only for me lol..

**Comment by Muhammad-Radwan** [comment]
@Detrav How can I print through a browser?

**Comment by Detrav** [comment]
[FastReport.OpenSource.Web](https://www.nuget.org/packages/FastReport.OpenSource.Web) package has a preview through the browser and, in theory, there should be a print button.

[Demo csproj](https://github.com/FastReports/FastReport/tree/master/Demos/OpenSource/FastReport.OpenSource.Web.MVC)

![image](https://user-images.githubusercontent.com/5013291/154437098-9e6f4579-1bb2-40e2-b781-4a4fe2f90b0a.png)

**Comment by Muhammad-Radwan** [comment]
@Detrav Can I use it in Desktop App?

**Comment by Detrav** [comment]
Through webview2 or CEF, maybe someone from the Fast Rerports team will agree to make a demo, I'm not sure.
Now you have to implement it yourself.

PS. The developers have a lot of work now, I don’t know when the bug with printing in the community will be fixed 😉

**Comment by Muhammad-Radwan** [comment]
Agreed my friend. Thanks a lot for the help.

**Comment by ilivecoding** [comment]
export to pdf
